RecoHGCal/TICL/plugins/TrackstersProducer.cc

```diff
@@ -0,0 +1,89 @@
+// Author: Felice Pantaleo - felice.pantaleo@cern.ch
+// Date: 09/2018
+// Copyright CERN
+
+// user include files
+#include <vector>
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+
+#include "RecoHGCal/TICL/interface/PatternRecognitionAlgoBase.h"
+#include "RecoHGCal/TICL/interface/Trackster.h"
+#include "RecoHGCal/TICL/plugins/TrackstersProducer.h"
+
+#include "PatternRecognitionbyCA.h"
+#include "PatternRecognitionbyMultiClusters.h"
+
+DEFINE_FWK_MODULE(TrackstersProducer);
+
+TrackstersProducer::TrackstersProducer(const edm::ParameterSet& ps)
+    : myAlgo_(std::make_unique<PatternRecognitionbyCA>(ps)) {
+  clusters_token_ = consumes<std::vector<reco::CaloCluster>>(
+      ps.getParameter<edm::InputTag>("hgcal_layerclusters"));
+  filtered_layerclusters_mask_token_ = consumes<std::vector<std::pair<unsigned int, float>>>(
+      ps.getParameter<edm::InputTag>("filtered_layerclusters_mask"));
+  original_layerclusters_mask_token_ =
+      consumes<std::vector<float>>(ps.getParameter<edm::InputTag>("original_layerclusters_mask"));
+  produces<std::vector<Trackster>>("TrackstersByCA");
```

Comment:
is there a plan to produce multiple collections from a single producer?
If not, please drop the instance specs